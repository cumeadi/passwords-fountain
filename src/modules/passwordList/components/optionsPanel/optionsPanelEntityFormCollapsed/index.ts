export { OptionsPanelEntityFormCollapsed } from './optionsPanelEntityFormCollapsed.component';
