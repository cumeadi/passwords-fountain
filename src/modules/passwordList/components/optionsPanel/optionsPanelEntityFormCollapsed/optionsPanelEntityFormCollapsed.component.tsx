import { h } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import PropTypes from 'prop-types';
import { Wrapper } from './optionsPanelEntityFormCollapsed.styles';
import { VariantProps } from '../optionsPanel.component';
import { optionsPanelVariantNames } from '@/modules/passwordList/passwordList.constants';
import { ButtonWrapper } from '../optionsPanel.styles';
import { Button } from '@/common/components/button';
import { Text } from '@/modules/localisation/components/text';
import { route } from 'preact-router';
import { useSelector } from '@preact-hooks/unistore';
import { selectIsInEditMode } from '@/modules/passwordList/passwordList.selectors';

export const OptionsPanelEntityFormCollapsed: TypedComponent<VariantProps> = ({
    switchCurrentVariantName,
}: VariantProps) => {
    const isInEditMode = useSelector(selectIsInEditMode);
    const actionLabel = isInEditMode
        ? 'optionsPanel.edit'
        : 'optionsPanel.addNew';

    const handleSettingsClick = (): void => {
        route('/settings');
    };
    const handleActionClick = (): void =>
        switchCurrentVariantName(optionsPanelVariantNames.entityFormExpanded);

    return (
        <Wrapper>
            <ButtonWrapper isInEditMode={isInEditMode}>
                <Button onClick={handleSettingsClick}>
                    <Text>optionsPanel.settings</Text>
                </Button>
                <Button onClick={handleActionClick}>
                    <Text>{actionLabel}</Text>
                </Button>
            </ButtonWrapper>
        </Wrapper>
    );
};

OptionsPanelEntityFormCollapsed.propTypes = {
    switchCurrentVariantName: PropTypes.func.isRequired,
};
