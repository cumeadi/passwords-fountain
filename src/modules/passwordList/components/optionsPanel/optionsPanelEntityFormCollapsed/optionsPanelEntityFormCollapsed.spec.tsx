import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { OptionsPanelEntityFormCollapsed } from './optionsPanelEntityFormCollapsed.component';
import { StoreProvider } from '@preact-hooks/unistore';
import createStore from 'unistore';

const stateMock = {
    passwordList: {
        selectedAndDecryptedEntity: {},
    },
};

describe('OptionsPanelEntityFormCollapsed', () => {
    it('should render correctly', () => {
        const { asFragment } = render(
            <StoreProvider value={createStore(stateMock)}>
                <OptionsPanelEntityFormCollapsed
                    switchCurrentVariantName={(): void => {
                        // TODO
                    }}
                />
            </StoreProvider>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
