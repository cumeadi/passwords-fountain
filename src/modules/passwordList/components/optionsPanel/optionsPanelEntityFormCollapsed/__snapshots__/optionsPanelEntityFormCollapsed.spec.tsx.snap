// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsPanelEntityFormCollapsed should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go115679"
  >
    <div
      class="go3306721962"
    >
      <button
        class="go3135434547"
        type="button"
      >
        <span
          class="go4062099715"
        >
          Settings
        </span>
      </button>
      <button
        class="go3135434547"
        type="button"
      >
        <span
          class="go4062099715"
        >
          Add new
        </span>
      </button>
    </div>
  </section>
</DocumentFragment>
`;
